torch/package/package_importer.py

```diff
@@ -402,6 +402,9 @@ def _handle_fromlist(self, module, fromlist, *, recursive=False):
                         self._handle_fromlist(module, module.__all__, recursive=True)
                 elif not hasattr(module, x):
                     from_name = "{}.{}".format(module_name, x)
+                    # handle case where from_name is ''
```

Comment:
I'm not sure why this change needs to be made-- I'm gonna guess it's because not all the parts that flow into/out of this function are the same as cpython's. The cpython's eventual logic calls `_find_and_load` which is different from ours, it could handle it there maybe. It's valid python to have the `fromlist=[""]` which is what triggers this case. It's not related to how the export's dunder import is written because the error is on the importing side not being able to resolve a package with a dot at the end:
```
__import__("subpackage_0.subsubpackage_0", globals(), fromlist=[""], level=1)
## Resulting error:
ModuleNotFoundError: No module named "package_b.subpackage_0.subsubpackage_0." in self-contained archive "<binary>" and the module is also not in the list of allowed external modules: ['']
```
The dot is added when the fromlist is parsed on line 404

Reply:
I figured this out, the `ModuleNotFoundError` should be suppressed by line 412 below, but it relies on a check against `exc.name`. 

Unfortunately, our implementation of the loader does not correctly populate the `name` attribute when raising a `ModuleNotFoundError`. 

The fix is to add `name=name` to the `ModuleNotFoundError` constructor on line 289.